components/TheHeader.vue

```diff
@@ -34,23 +47,44 @@ export default {
   },
   data() {
     return {
-      searchTask: '',
+      search: false,
+      searchText: '',
+      debounced: null,
     };
   },
   computed: {
     ...mapGetters({
       locales: 'lang/getLocals',
       currentLocale: 'lang/getCurrentLocale',
+      isSearching: 'todos/getIsSearching',
+      showSearchInput: 'todos/getShowSearchField',
     }),
   },
   mounted() {
     this.$i18n.setLocale(this.currentLocale.code);
+    this.debounced = debounce(this.searchTask, 500);
   },
   methods: {
     switchLanguage(event) {
       this.$store.dispatch('lang/setLocale', event.target.value);
       this.$i18n.setLocale(event.target.value);
     },
+    async searchTask() {
+      this.$store.dispatch('todos/setIsSearching', true);
+      await this.$store.dispatch('todos/setSearchText', this.searchText);
+      this.$store.dispatch('todos/filterTaskList');
+      this.$store.dispatch('todos/setIsSearching', false);
+      this.$store.dispatch('todos/resetLimit');
+      this.$store.dispatch('todos/setTotalPage');
+    },
+    async setSearch() {
+      this.search = !this.search;
+      await this.$store.dispatch('todos/setShowSearchField', this.search);
```

Comment:
why do we need await?

Reply:
I am returning a promise here that showSearchField is set to  true  because only then that input field will be available in DOM and we will be able to change it.

Reply:
await removed to solve the focus issue ,$nextTick used instead